liquidator/bam_scorer.h

```diff
@@ -179,6 +198,58 @@ class BamScorer
     }
 
 private:
+
+    struct BamAllocator
+    {
+        BamAllocator() { }
+
+        ~BamAllocator()
+        {
+            // note: we are not calling bam_destroy1(), which was just free()ing the
+            // bam and the bam data. since the vectors manage the bam memory,
+            // we just need to free the bam dat, but if samtools changes this will
+            // likely break
+            free(bam.data);
```

Comment:
This seems like a good trade off given that we are so concerned with performance.